src/rextester.js

```diff
@@ -0,0 +1,137 @@
+const languageProperties = require('./languageProperties.json');
+const superagent = require('superagent')
+
+function parseCode(code, languageObject, imports) {
+    if (languageObject.classDeclaration) { //Add class declaration and replace {{code}} with the user's code
+        let classDec = languageObject.classDeclaration
+        classDec = classDec.replace("{{code}}", code)
+        code = classDec
+    }
+    let i = 0
+    imports.forEach(element => { //If import already exists in defaults then remove it from the array
+        if (languageObject.defaultImports.includes(element)) {
+            imports.splice(i, 1)
+        }
+        i++
+    });
+    languageObject.defaultImports.forEach(element => {  //Add the default imports the the imports array
+        imports.push(element)
+    })
+    imports.forEach(element => { //Add the imports to the top of the code
+        code = `${languageObject.importType} ${element}${languageObject.lineBreak}\n${code}`
+    });
+    return code
+}
+
+function parseLanguages(args, imports) { //Add the imports to the array that were user-added
+    for (let i = 2; i < args.length; i++) {
+        if (args[i].endsWith(';')) {
+            imports.push(args[i].substr(0, args[i].length - 1));
+            i = args.length + 1;
+        } else {
+            imports.push(args[i]);
+        }
+    };
+}
+
+async function outputResult(language, code, compilerArgs, callback) {
+    let r;
+    let response = await superagent
+        .post('https://rextester.com/rundotnet/api')
+        .set({
+            "Content-Type": "application/json"
+        })
+        .send({
+            LanguageChoice: language,
+            Program: code,
+            Input: "",
+            CompilerArgs: compilerArgs
+        })
+        .catch((err) => r = {
```

Comment:
Since you are using await, it would maybe clearer to wrap the request in a try catch (pure code style).
However what you wrote here is totally valid and I don't mind it.